pkg/scaffold/cmd.go

```diff
@@ -90,15 +90,24 @@ func main() {
 	}
 
 	// Become the leader before proceeding
-	leader.Become(context.TODO(), "{{ .ProjectName }}-lock")
+	err = leader.Become(context.TODO(), "{{ .ProjectName }}-lock")
+	if err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
 
 	r := ready.NewFileReady()
 	err = r.Set()
 	if err != nil {
 		log.Error(err, "")
 		os.Exit(1)
 	}
-	defer r.Unset()
+	defer func() {
+		if err = r.Unset(); err != nil {
```

Comment:
We should consider instead to remove the error check from `Unset` and instead allow it to panic if it errors IMO. 
/cc @mhrivnak

Reply:
I'm not a fan of panics in general, and I subscribe to the philosophy articulated [here](https://github.com/golang/go/wiki/PanicAndRecover#usage-in-a-package). Printing a stack trace of each goroutine is usually not that helpful to an end user.

Logging the error and exiting with a non-zero code seems like the best thing to do in this circumstance.